updater/updater.go

```diff
@@ -63,8 +75,9 @@ func (t *Target) StreamTo(suffix string, r io.Reader) error {
 	} else {
 		hash = sha256.New()
 	}
-	var cw countingWriter
-	req, err := http.NewRequest(http.MethodPut, t.BaseURL+"update/"+suffix, io.TeeReader(io.TeeReader(r, hash), &cw))
+	cw := &countingWriter{progress: t.progress}
+	defer close(t.progress)
+	req, err := http.NewRequest(http.MethodPut, t.BaseURL+"update/"+suffix, io.TeeReader(io.TeeReader(r, hash), cw))
```

Comment:
Thinking about it, maybe the caller of `StreamTo` could just supply an `io.TeeReader()` as `r` and keep track of written bytes and hence progress themselves?

Channels in APIs are easy to get wrong on the consuming side, so I’m cautious about adding them.